### PR TITLE
Add Studio Code and Director (and some more) to JavLibrary_python

### PR DESCRIPTION
**Pull request by YurikaL**
Changes:

- Add new field that recently added to Stash: Studio Code and Director
- Reorganize fields: move DVD code from Title to actual Code, move scene title from Details to actual Title
- Add option to strip (or not) DVD code from Title
- Disable Details parsing since there are no real details in JavLibrary
- Add Label and Performer name order. This features are not supported by Stash yet, but why not implement it in advance? JAV progress in Stash tracked in stashapp/stash#3065
- Reverse Performer name order only if explicitly selected. Set Western name order for Japanese performers is ~~always a silly idea~~ not always preferable, so it better to have option to control that.

### CODE REVIEW

**Comment by DogmaDragon** [comment]
Good things to consider: 
- StashDB requires Western order for primary names. <https://guidelines.stashdb.org/docs/performers/performer-names-and-aliases/#jav-performer-names>
- My understanding is that on StashDB JAV DVD codes are used as titles for a reason. 

Defaults really matter when it comes to scrapers since that's what gets submitted most of the time.

**Comment by YurikaL** [comment]
- If StashDB claims that it does stuff the same way as MusicBrainz, then there is only one correct option to store everything: identify each item by UUID, and store each relevant data piece within its own field, without obligatory distinction which field is primary and which is secondary. Where is the best place to submit this issue?
 - Using DVD code as titles was indeed reasonable for JAV as long as there no separate Code field was available, but now, since 0.18 release, there is. In my opinion, UI should be changed to display Code **and** Title where applicable. Where is the best place to submit this issue?
 - What do you suggest for properly storing JAV DVD codes? Are you assuming that now there should be two different fields with exactly the same content? Are you implying that actual Details should keep being rejected and/or blended with actual Title (and Series)?
 - Is StashDB even a thing? Since it unavoidable requires joining a certain get-together and begging for invite - I don't think so. When a public release planned? Well, maybe StashDB team wants to have a well-polished and fine-tuned data format and code before a public release to prevent an unwanted mess? So it would be nice to mass-move JAV DVD codes from Title to actual Code. Where is the best public place to submit this suggestion?

**Comment by DogmaDragon** [comment]
> * If StashDB claims that it does stuff the same way as MusicBrainz, then there is only one correct option to store everything: identify each item by UUID, and store each relevant data piece within its own field, without obligatory distinction which field is primary and which is secondary. Where is the best place to submit this issue?

https://github.com/stashapp/stash-box/issues
> * Using DVD code as titles was indeed reasonable for JAV as long as there no separate Code field was available, but now, since 0.18 release, there is. In my opinion, UI should be changed to display Code **and** Title where applicable. Where is the best place to submit this issue?

https://github.com/stashapp/stash-box/issues
> * What do you suggest for properly storing JAV DVD codes? Are you assuming that now there should be two different fields with exactly the same data? Are you implying that actual Details should keep being rejected and/or blended with actual Title (and Series)?

Not up to me to decide what is what. I'm just raising questions I have. 
> * Is StashDB even a thing? Since it unavoidable requires joining a certain get-together and begging for invite - I don't think so. When a public release planned? Well, maybe StashDB team wants to have a well-polished and fine-tuned data format and code before a public release to prevent an unwanted mess? So it would be nice to mass-move JAV DVD codes from Title to actual Code. Where is the best public place to submit this suggestion?

https://github.com/stashapp/stash-box/issues

**Comment by bnkai** [comment]
>If StashDB claims that it does stuff the same way as MusicBrainz, then there is only one correct option to store everything: identify each item by UUID, and store each relevant data piece within its own field, without obligatory distinction which field is primary and which is secondary. Where is the best place to submit this issue?

Please join the official discord channel and make a new thread or post in the relevant sections there. Stashdb guidelines live in the discord server

>Using DVD code as titles was indeed reasonable for JAV as long as there no separate Code field was available, but now, since 0.18 release, there is. In my opinion, UI should be changed to display Code and Title where applicable. Where is the best place to submit this issue?

UI needs to display both where exactly? Do you mean in `stash's` scenes page ? If it is purely related to stash then a request in stash's github page is the proper place.

>What do you suggest for properly storing JAV DVD codes? Are you assuming that now there should be two different fields with exactly the same data? Are you implying that actual Details should keep being rejected and/or blended with actual Title (and Series)?

For the majority of titles mentioned the Details/Title/Series part is already messed up. Even now with the existence of the code field the `titles` (that we now add as details) from javlibrary are too long for my personal preference. I would prefer to keep the code as a title but that's just me.
For the `actual` details do you have a sample url or something, most movies i see dont have any? We could (for now at least) concat the `current` with the `actual` details
 In general we try to stick to what the stashdb uses by `default`. I would use the current defaults and add an option or new selectors commented out for the title\details\series\... you mention. Anyway for the defaults it is better if you join the discord server to raise the issue in the #stashb-guidelines channel

>Is StashDB even a thing? Since it unavoidable requires joining a certain get-together and begging for invite - I don't think so. When a public release planned? Well, maybe StashDB team wants to have a well-polished and fine-tuned data format and code before a public release to prevent an unwanted mess? So it would be nice to mass-move JAV DVD codes from Title to actual Code. Where is the best public place to submit this suggestion?
Stashb for mainstream content works very well and it has been up for some time. There is no `begging` involved. You just join stash's discord server and get an invite from the pinned message in the respective channel. If you want to have a quick look <https://stashapp.github.io/Stash-Docs/docs/Beginner-Guides/Guide-To-Scraping/> shows how more effective it is to use stashdb for your scraping.

To sum it up for this PR 
- please keep the current title and performer naming as the default (at least till we decide on guidelines)
- add as an extra option or in commented code your own preferences
- for the details if you have them concatanate to the existing ones

**Comment by YurikaL** [comment]
> Please join the official discord channel 

This is not an option for me. Please consider establishing a public discussion platform. I'll post my suggestions in the Discussions on github, and then it's up to you.

> UI needs to display both where exactly? Do you mean in stash's scenes page ? If it is purely related to stash then a request in stash's github page is the proper place.

I don't know whether it is related only to Stash itself or Stash-Box/StashDB/StashWhatever too. Anyway, thanks for answer.

> Even now with the existence of the code field the titles (that we now add as details) from javlibrary are too long for my personal preference. I would prefer to keep the code as a title but that's just me.

Again, in my opinion it is a kind of UI task to truncate titles if they are too long (and show the full version on mouse over). Another issue to fill, I see.

> For the actual details do you have a sample url or something, most movies i see dont have any?

You are correct, quite a small number of movies had details in R18. Here are some rare examples:
https://web.archive.org/web/20171230115804/http://www.r18.com/videos/vod/movies/detail/-/id=bban00010/
https://web.archive.org/web/20180708052336/http://www.r18.com/videos/vod/movies/detail/-/id=lzkw00003/

On Fanza, every or almost every movies has details, but of course in Japanese. Ideally, there should be an amalgamating JAV scrapper that collect JAV metadata from all possible places, something like [JAVMovieScraper](https://github.com/DoctorD1501/JAVMovieScraper), [Javinizer](https://github.com/jvlflame/Javinizer) or even [jellyfin-plugin-javtube](https://github.com/javtube/jellyfin-plugin-javtube/wiki/数据来源). Or native amalgamating multiscraper within Stash. Which way is preferable and more realistic, in your opinion?

> There is no begging involved. You just join stash's discord server and get an invite from the pinned message in the respective channel.

Got it, but I still encourage StashDB team to open public access with no certain private chats involved. 

> - please keep the current title and performer naming as the default (at least till we decide on guidelines)
> - add as an extra option or in commented code your own preferences
> - for the details if you have them concatanate to the existing ones

*Naruhodo ne.* Legacy naming is now kept by default, extra options added, title is concatenated to details.

**Comment by bnkai** [comment]
> On Fanza, every or almost every movies has details, but of course in Japanese. Ideally, there should be an amalgamating JAV scrapper that collect JAV metadata from all possible places, something like [JAVMovieScraper](https://github.com/DoctorD1501/JAVMovieScraper), [Javinizer](https://github.com/jvlflame/Javinizer) or even [jellyfin-plugin-javtube](https://github.com/javtube/jellyfin-plugin-javtube/wiki/%E6%95%B0%E6%8D%AE%E6%9D%A5%E6%BA%90). Or native amalgamating multiscraper within Stash. Which way is preferable and more realistic, in your opinion?


IMHO it would be easier to just add the meta in stashdb (via exisitng scrapers and maybe a few new if we dont have existing for some of the sites) and then use the tagger interface to scrape from there directly.
Any PRs are welcome though.
